src/main/java/at/hannibal2/skyhanni/utils/TimeOffset.kt

```diff
@@ -0,0 +1,56 @@
+package at.hannibal2.skyhanni.utils
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.DebugDataCollectEvent
+import at.hannibal2.skyhanni.events.ProfileJoinEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import kotlinx.coroutines.launch
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.apache.commons.net.ntp.NTPUDPClient
+import java.net.InetAddress
+import kotlin.math.abs
+import kotlin.math.absoluteValue
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+@SkyHanniModule
+object TimeOffset {
+    var offsetMillis = 0L
+        private set
+
+    init {
+        SkyHanniMod.coroutineScope.launch {
+            offsetMillis = getNtpOffset("time.google.com") ?: 0
+            print("Time offset: $offsetMillis")
+        }
+    }
+
+    private fun getNtpOffset(ntpServer: String): Long? {
+        return try {
+            val client = NTPUDPClient()
+            val address = InetAddress.getByName(ntpServer)
+            val timeInfo = client.getTime(address)
+
+            timeInfo.computeDetails()
+            timeInfo.offset
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
please throw a skyhanni error in chat

Reply:
this runs during game boot, I'm pretty sure the user wouldn't even see the error

Reply:
> this runs during game boot, I'm pretty sure the user wouldn't even see the error

this is a correct assumption